### PR TITLE
Python federated

### PR DESCRIPTION
**Pull request by Soroosh129**
This adds support for local federated programs written using the Python target. The data exchanged between federates is transparently serialized and deserialized using Python's pickle module.

It also adds a simple HelloDistributed.lf test.

### CODE REVIEW

**Comment by housengw** [comment]
#703

**Comment by Soroosh129** [comment]
I think this is ready to be merged. There are a few failing federated tests but fixing them requires augmenting the Python API to add a few missing functionalities. These changes can be put in their own PR to avoid making this PR larger than it already is. I suggest we could ignore the tests in the `failing` folder for now during the review process.

Thanks @housengw for meticulously porting all the federated C tests to Python. I am optimistic we can get them all passing.

File: org.lflang/src/org/lflang/validation/LFValidator.xtend [hunk 1]
```diff
@@ -83,7 +83,7 @@ import org.lflang.lf.WidthSpec
 
 import static extension org.lflang.ASTUtils.*
 import static extension org.lflang.JavaAstUtils.*
-import org.lflang.federated.SupportedSerializers
+import org.lflang.federated.serialization.SupportedSerializers
```
Comment (lhstrh) [inline org.lflang/src/org/lflang/validation/LFValidator.xtend:86]: Should serialization be part of the `federated` package? As you mentioned this morning, we might want to employ serialization to enable communication between reactors of different targets without relying to inter-process communication.

Comment (Soroosh129) [reply org.lflang/src/org/lflang/validation/LFValidator.xtend:86]: That's an excellent point.

I think for the time-being, the serialization package is mainly focused on federated execution. Once we are able to generalize its functionality, we should definitely consider moving it out of the federated package.

**Review by lhstrh — APPROVED** [review]
Looks good! Should we first merge https://github.com/lf-lang/lingua-franca/pull/713?

**Comment by Soroosh129** [comment]
> Looks good! Should we first merge #713?

I think the order doesn't matter (hopefully). If we merge this in first, we can change the base of #713 to master.

**Pull request merged**